test/unit/mdc-textfield/mdc-text-field-bottom-line-foundation.test.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {assert} from 'chai';
+import td from 'testdouble';
+
+import {verifyDefaultAdapter} from '../helpers/foundation';
+import {setupFoundationTest} from '../helpers/setup';
+import MDCTextFieldBottomLineFoundation from '../../../packages/mdc-textfield/bottom-line/foundation';
+
+const {cssClasses} = MDCTextFieldBottomLineFoundation;
+
+suite('MDCTextFieldBottomLineFoundation');
+
+test('exports cssClasses', () => {
+  assert.isOk('cssClasses' in MDCTextFieldBottomLineFoundation);
+});
+
+test('defaultAdapter returns a complete adapter implementation', () => {
+  verifyDefaultAdapter(MDCTextFieldBottomLineFoundation, [
+    'addClass', 'removeClass', 'setAttr',
+    'registerEventHandler', 'deregisterEventHandler',
+    'notifyAnimationEnd',
+  ]);
+});
+
+const setupTest = () => setupFoundationTest(MDCTextFieldBottomLineFoundation);
+
+test('#init adds event listeners', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.init();
+
+  td.verify(mockAdapter.registerEventHandler('transitionend', td.matchers.isA(Function)));
+});
+
+test('#destroy removes event listeners', () => {
+  const {foundation, mockAdapter} = setupTest();
+  foundation.destroy();
+
+  td.verify(mockAdapter.deregisterEventHandler('transitionend', td.matchers.isA(Function)));
+});
+
+test('activate adds mdc-text-field__bottom-line--active class', () => {
```

Comment:
Use a template string to plug in `cssClasses.BOTTOM_LINE_ACTIVE` here and in the next test name, like you did for the animation end event test elsewhere?